pkg/odo/cli/registry/registry.go

```diff
@@ -0,0 +1,214 @@
+	listCmd.Flags().StringVar(&o.devfileRegistryFlag, "devfile-registry", "", "Only show components from the specific Devfile registry")
+	listCmd.Flags().BoolVar(&o.detailsFlag, "details", false, "Show details of each component")
+
+	// Add a defined annotation in order to appear in the help menu
+	listCmd.Annotations["command"] = "main"
+	listCmd.SetUsageTemplate(odoutil.CmdUsageTemplate)
+
+	return listCmd
+}
+
+func (o *ListOptions) printDevfileList(DevfileList []registry.DevfileStack) {
+
+	// Create the table and use our own style
+	t := table.NewWriter()
+
+	t.SetStyle(table.Style{
+		Box: table.BoxStyle{
+			PaddingLeft:  " ",
+			PaddingRight: " ",
+		},
+		Color: table.ColorOptions{
+			Header: text.Colors{text.FgHiGreen, text.Underline},
+		},
+		Format: table.FormatOptions{
+			Footer: text.FormatUpper,
+			Header: text.FormatUpper,
+			Row:    text.FormatDefault,
+		},
+		Options: table.Options{
+			DrawBorder:      false,
+			SeparateColumns: false,
+			SeparateFooter:  false,
+			SeparateHeader:  false,
+			SeparateRows:    false,
+		},
+	})
+	t.SetOutputMirror(log.GetStdout())
+
+	t.AppendHeader(table.Row{"NAME", "REGISTRY", "DESCRIPTION"})
+	t.SortBy([]table.SortBy{
+		{Name: "NAME", Mode: table.Asc},
+		{Name: "REGISTRY", Mode: table.Asc},
+	})
+
+	for _, devfileComponent := range DevfileList {
+
+		// Mark the name as yellow in the index to it's easier to see.
+		name := text.Colors{text.FgHiYellow}.Sprint(devfileComponent.Name)
+
+		// If the user has specified a filter with variable o.filterFlag, then only show the components
+		// containing that specific string.
+		if o.filterFlag != "" {
+			if !strings.Contains(devfileComponent.Name, o.filterFlag) {
+				continue
+			}
+		}
+
+		// If the user passed in --devfile-registry <REGISTRY-NAME>, then only show the components from that Devfile stack
+		if o.devfileRegistryFlag != "" {
+			if !strings.Contains(devfileComponent.Registry.Name, o.devfileRegistryFlag) {
+				continue
+			}
+		}
+
+		// If the user passed in --devfile <NAME> only show that specific component matching that name
+		if o.devfileFlag != "" {
+			if devfileComponent.Name != o.devfileFlag {
+				continue
+			}
+		}
+
+		if o.detailsFlag {
+
+			fmt.Printf(`Name: %s
+Display Name: %s
+Registry: %s
+Registry URL: %s
+Version: %s
+Description: %s 
+Tags: %s
+Project Type: %s
+Language: %s
+Starter Projects:
+  - %s
+Supported odo features:
+  - dev: %s
+  - deploy: %s
+  - debug: %s
+%s`,
+				devfileComponent.Name,
+				devfileComponent.DisplayName,
+				devfileComponent.Registry.Name,
+				devfileComponent.Registry.URL,
+				devfileComponent.Version,
+				devfileComponent.Description,
+				strings.Join(devfileComponent.Tags[:], ", "),
+				devfileComponent.ProjectType,
+				devfileComponent.Language,
+				strings.Join(devfileComponent.StarterProjects, "\n  - "),
+				// TODO, showing dev / deploy / debug NOT yet implemented
```

Comment:
If there are things that are not implemented yet, and you plan to add it in this PR, please mark PR as work in progress.

Reply:
@cdrage this part is still commented. Is it going to be handled in this PR or a subsequent one?